src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,54 @@
+.calendar {
+  display: flex;
+  justify-content: center;
+  padding: $field-paddings;
+  width: $calendar-field-width;
+
+  @for $i from 28 through 31 {
+    &--month-length--#{$i} {
+      .calendar__cell:nth-child(n + #{$i+1}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $name, $number in $days {
+    &--start-day--#{$name} {
+      .calendar__cell:first-child {
+        margin-left: ($number - 1) * ($cell-size + $gap-size);
+      }
+    }
+  }
+
+  &__field {
+    display: flex;
+    flex-wrap: wrap;
+    gap: $gap-size;
+
+    max-width: $calendar-field-width;
+  }
+
+  &__cell {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    @include cell($cell-size);
+
+    background-color: $cell-color;
+    border: 1px solid $border-color;
+
+    @for $i from 1 through 31 {
+      &:nth-child(n+#{$i})::after() {
+        content: "#{$i}";
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: $cell-focus-color;
+      transform: translateY(-20px);
+      transition: 0.5s;
```

Comment:
don't animate all, specify properties you need

Reply:
WDYM

Reply:
Yes, it is generally better to specify only the properties that need to be animated in the CSS transition property. This can help to optimize the performance of the animation and reduce unnecessary processing.

In the code snippet provided, only the transform property and the transition property are being animated. Therefore, it would be better to specify only those properties in the transition property, like this:
```
transform: translateY(-20px);
transition: transform 0.5s;

```

This will ensure that only the transform property is animated with a duration of 0.5 seconds. The background-color property will not be affected by the animation.

Reply:
that's ok. but...

Reply:
that's ok, but... The task is:

`The hovered cell has to become pink
  - (**Optional**) Move it up by `20px` (use `transform`)
  - (**Optional**) both of these properties are to be animated with the duration of half a second.
`